Update aura effects names

T1ti opened the pull request:
Replace aura effects names by clear text instead of the enums
The first half comes from official blizzzard naming in a dbc

before | after
![image](https://user-images.githubusercontent.com/40864460/140400818-9eb6b1f9-2346-43a6-a9f9-36ce86af4975.png)

T1ti commented:
There's a design choice for the "mod vs creature type" effects, the official Blizzard naming is "Mod creature X" like **"Mod Creature Attack power"**. Which means player has increased AP vs this type of creature

I didn't think it was very descriptive so I changed it to "Mod Ranged Attack Power Versus Creature", it's a lot longer but more descriptive

stoneharry commented:
My thoughts are that this is a lot friendlier for most users, and it really helps with scanning down the list seeing immediately the relevant information rather than everything being prefixed with `SPELL_AURA_`.

However programmers would often prefer the original enum name - having the full enum name makes it a lot easier to search in the emulator for where it is used. If you just have the enum ID then unless you know where to look it can be hard to find the definition.

To try and please both users there are a few different approaches we could take:

1) Doing a better formatted label

Currently it is a string that is used in 4 combo boxes. We could build a panel that has formatted labels to show the ID, the friendly name, and the enum.

This might look pretty but wewould need to construct unique panels for each of the 4 combo boxes that use these values.

2) Having a developer and friendly language files.

We could have `enUS - Friendly` and `enUS - Developer` language definitions. This would allow you to switch the display with the dropdown on the top right. The downsides is that it is more language definitions to maintain.

3) Just display both

We could display both the friendly name and the enum value on the same line and just have it as a long string. We would have to do some whitespace padding to make it look like the two values are in columns.

What do you think?

darrian12 commented:
>3) Just display both

This _might_ end up looking chaotic.
>4) Adding tooltips

It sounds nice, but could be bothersome if you want to find a specific Enum. 
> 1) Doing a better formatted label

Reasonable solution.
<hr>

>2) Having developer and friendly language files

I like this one the most. Because it's not chaotic, allows you to toggle to either display and has a clear seperation. Even though it's more to maintain it's also something that can be expanded on a lot more in the future. For example if you wanted to make the editor even more friendly you can very easily build upon this while still allowing the user to have that freedom between choosing either.

T1ti commented:
I think the enum should be displayed either only on tooltip or by enabling a dev mode
so 2 or 4 for me

stoneharry commented:
@T1ti I have pushed the tooltip solution to my copy of this pull request: https://github.com/stoneharry/WoW-Spell-Editor/commit/1fa602dfbbe8eeeed73fe4b2a26f4b90afe73cf5

If you like the changes can you go ahead and merge them into your branch and push it up to this pull request please? Once you have done that I think we can merge the PR. :)

T1ti commented:
@stoneharry I forgot I used the same branch so I overrided this PR with all my local changes

It includes : 

-  UI rework of the base tab with categories

![image](https://user-images.githubusercontent.com/40864460/144760866-7de1fa16-e38b-449f-bdc3-5072d1087de9.png)

- Addition of selectors for Global cd category, spell family name
- updated the category names from official DBC files 
- 
![image](https://user-images.githubusercontent.com/40864460/144760983-178aa4c9-d8f5-46e8-bccf-3232b1accaeb.png)

- made range selector less busy by merging ally/enemy when they are the same 
![image](https://user-images.githubusercontent.com/40864460/144761018-efbbe975-8189-4a09-8046-494fd7c7f641.png)

- added misc values selectors for almost all aura types
- 
![image](https://user-images.githubusercontent.com/40864460/144761039-24393080-c0aa-47a3-b362-814baa16fe82.png)
![image](https://user-images.githubusercontent.com/40864460/144761042-21d55da1-40d1-4025-b726-c177746c453a.png)
https://i.gyazo.com/16c2bec1632f890e3c8fcbd9e174eb06.mp4 and even more of them. They're just implemented, not properly placed, and only spell effect 1 is done.

lapizwcg commented:
How to download this modified version?

stoneharry commented:
> How to download this modified version?

You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.

lapizwcg commented:
> > How to download this modified version?
> 
> You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.

Sorry, I'm very weak in programming, I don't know how to do it at all. Can you please put together a ready-to-use version with all the changes?

lapizwcg commented:
If you can, please compile the **T1ti** version into a complete one and upload it somewhere where you can download it. I will be very, very grateful

lapizwcg commented:
here is my DS Sioz#1270

lapizwcg commented:
> > How to download this modified version?
> 
> You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.

Please help me to make programm with t1ti changes)))

stoneharry commented:
> > > How to download this modified version?
> > 
> > 
> > You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.
> 
> Please help me to make programm with t1ti changes)))

Please do not keep commenting the same thing.

What exactly is your use case? Why do you want these changes?

lapizwcg commented:
> > > > How to download this modified version?
> > > 
> > > 
> > > You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.
> > 
> > 
> > Please help me to make programm with t1ti changes)))
> 
> Please do not keep commenting the same thing.
> 
> What exactly is your use case? Why do you want these changes?

because it is easier to perceive information when it is written in understandable language.  for example "SPELL_AURA_MOD_DAMAGE_DONE", but normally it's just "+ weapon damage". maybe there is a DISCORD channel, with a discussion of this program, where you can contact T1ti or whoever continues to develop the program?  unfortunately I could not build the program using VISUAL STUDIO

stoneharry commented:
> > > > > How to download this modified version?
> > > > 
> > > > 
> > > > You would have to pull in this branch and compile it yourself. There are too many changes thrown into this PR to review it very easily, it will probably remain an inactive branch until we work out how to break it up and cherry-pick from it.
> > > 
> > > 
> > > Please help me to make programm with t1ti changes)))
> > 
> > 
> > Please do not keep commenting the same thing.
> > What exactly is your use case? Why do you want these changes?
> 
> because it is easier to perceive information when it is written in understandable language. for example "SPELL_AURA_MOD_DAMAGE_DONE", but normally it's just "+ weapon damage". maybe there is a DISCORD channel, with a discussion of this program, where you can contact T1ti or whoever continues to develop the program? unfortunately I could not build the program using VISUAL STUDIO

I am the maintainer of this program. There is no Discord channel for it. I track discussions on this program this the Github project: through issues and pull requests, like we are doing now.

This PR was originally about only updating the aura names, one field on the entire UI. You can see in this thread there is some discussion on this and the advantages/disadvantages. Making aura names 'easier to read' is not always desired - there are different audiences and this is discussed here.

This was ready to merge but Titi added tons of other unrelated changes to the branch. This is too much mish-mashed content to review and merge together, it needs to be split up and discussed individually.

I can spin up a pull request from my branch that handles just the aura renaming at some point, but I don't have a lot of free time at the moment.

Luckily you can actually change this without recompiling the program thanks to the language XAML feature. You can download this single file and place it in the languages folder for the spell editor: https://raw.githubusercontent.com/stoneharry/WoW-Spell-Editor/1fa602dfbbe8eeeed73fe4b2a26f4b90afe73cf5/Documentation/Languages/enUS.xaml

T1ti commented:
final version in https://github.com/stoneharry/WoW-Spell-Editor/pull/255 without much extra fluff
